public/js/components/Editor.js

```diff
@@ -101,12 +98,18 @@ const Editor = React.createClass({
     const closePanel = () => this.cbPanels[line].clear();
 
     if (isEnabled("conditionalBreakpoints") && bp && ev.metaKey) {
-      const { condition } = bp;
-      const panel = renderConditionalBreakpointPanel({
-        location, setBreakpointCondition, condition, closePanel
-      });
-
-      this.cbPanels[line] = this.editor.codeMirror.addLineWidget(line, panel);
+      if (!this.state.isCondBPOpen) {
```

Comment:
Hmm, I'm not sure if this would work, but it might be nice to try getting rid of the two new state fields: `isCondBPOpen` and `openPanel`.

I think we can do that because, `this.cbPanels[line]` should have all the information we need. e.g.

`this.state.isCondBPOpen == !! this.cbPanels[line]`
`this.openPanel == this.cbPanels[line]`